### ISSUE TITLE
quality: enforce config key validation, complete RNG seeding, and add config hash to PPO training

### ISSUE THREAD

**Issue by ll7** [ll7/robot_sf_ll7#6483]
## Problem

The training scripts have inconsistent config validation, RNG seeding, and provenance recording, creating reproducibility risks.

### 1. PPO training script silently ignores unknown config keys (High)

**File:** `scripts/training/train_ppo.py:930-1050`

`load_expert_training_config` reads YAML via `data.get(...)` but never checks for unexpected keys. Four other training scripts (`train_sac_sb3.py`, `train_distributional_rl.py`, `train_constrained_rl.py`, `train_dreamerv3_rllib.py`) all reject unknown keys. A typo like `total_timestep:` (missing 's') silently falls through.

**Fix:** Add `_ALLOWED_ROOT_KEYS` frozenset and reject unknown keys, consistent with the other four scripts.

### 2. Incomplete RNG seeding (Medium)

**Files:** `scripts/training/train_distributional_rl.py:397-398`, `scripts/training/train_predictive_planner.py:759-760`

- Distributional RL seeds `random` and `torch` but NOT `numpy`.
- Predictive planner seeds `numpy` and `torch` but NOT `random`.

The canonical `set_global_seed` in `robot_sf/common/seed.py:133` seeds all three plus CUDA. Neither script uses it.

**Fix:** Replace ad-hoc seeding with `common.set_global_seed(config.seed)`.

### 3. No config hash in PPO training script (Medium)

**File:** `scripts/training/train_ppo.py` (3419 lines)

Zero matches for `config_hash`, `sha256.*config`. The script copies the config YAML next to the checkpoint but computes no integrity hash. By contrast, `run_predictive_training_pipeline.py:593` computes SHA-1 and `compare_issue_4014_ppo_sequence_models.py:90` computes SHA-256 — no consistent convention.

**Fix:** Compute SHA-256 of resolved config at load time, include in startup log and checkpoint metadata. Standardize on SHA-256.

## Acceptance criteria

- [ ] `train_ppo.py` rejects unknown config keys.
- [ ] All training scripts use `set_global_seed` from `robot_sf/common/seed.py`.
- [ ] `train_ppo.py` records a SHA-256 config hash in its manifest.
- [ ] `uv run pytest tests` passes.

## Evidence

Code review on 2026-07-30.

**Comment by ll7** [comment]
Audit application — first bounded child: #6489 owns strict, side-effect-free unknown-key rejection in `train_ppo.py`. RNG seeding and config-hash provenance remain separate. Duplicate search found no equivalent issue or PR. Decision source: Codex task 019fb13c-6ea8-7632-a41b-f9940898a48c.

**Comment by ll7** [comment]
factory-v2: split into bounded child issues; parent remains coordination-only

**Comment by ll7** [comment]
### Blocked-queue triage (tracking #7067)



```yaml
blocker_class: upstream_issue
unblock_condition: >-
  The bounded children covering unknown-key rejection (#6489), complete RNG seeding, and config-hash provenance are all closed.
watcher: >-
  `gh issue list --search 'is:open 6483 in:body'` returns no open children
next_check_at: >-
  2026-09-01
last_meaningful_progress_at: "2026-07-30"
```

Converted to coordination-only on 2026-07-30; #6489 owns the first slice and the remaining two findings are separate.

Bookkeeping only — no scope, claim, label, or evidence change is made by this comment, and the issue stays open.

**Comment by ll7** [comment]
## Unblock route — 2026-08-16

This is local implementation work. Reproduce the current defect on `main`, freeze the smallest failing fixture, patch the canonical owner only, and add a regression that proves both the corrected path and the existing fail-closed path. Avoid coupling the repair to campaign execution or unrelated refactors.

**Comment by ll7** [comment]
## Verified complete

The current `origin/main` (`ccca0f49d8136948d3702bba1760540fe8a487c3`) already satisfies this issue through merged child fixes; no duplicate implementation is needed:

- Unknown PPO configuration keys are rejected before launch (`0ad1fdb50`, issue/PR #6562).
- SHA-256 hashing is computed from the resolved PPO config, logged, and included in the training-run manifest (`e6310a718`, issue #6500 / PR #6568).
- Distributional-RL and predictive-planner training use `robot_sf.common.set_global_seed`, which covers Python, NumPy, and PyTorch (`52a6186c9`, issue/PR #6563); PPO uses the same helper for its configured seed.
- Focused verification in a fresh linked worktree passed: `49 passed` for `tests/training/test_expert_ppo_config_key_validation.py`, `tests/training/test_train_seeding.py`, and `tests/training/test_train_expert_ppo_contract.py`.
- The descendant repository readiness run also passed all current CI lanes (29 successful, one intentional coverage skip).

This parent issue was stale after the child fixes landed, so it is being closed as completed. The existing acceptance contract and evidence boundary remain unchanged.
